mlir/lib/QEC/Utils/PauliStringWrapper.cpp

```diff
@@ -79,12 +78,19 @@ bool PauliStringWrapper::commutes(const PauliStringWrapper &other) const
 PauliStringWrapper
 PauliStringWrapper::computeCommutationRulesWith(const PauliStringWrapper &rhs) const
 {
-    // P * P' * i
-    stim::FlexPauliString result =
-        (*this->pauliString) * (*rhs.pauliString) * stim::FlexPauliString::from_text("i");
-
+    stim::FlexPauliString result = *rhs.pauliString;
+    assert(llvm::isa<PPRotationOp>(this->op) && "Clifford Operation is not PPRotationOp");
+    if (auto this_op = llvm::cast<PPRotationOp>(this->op); this_op.hasPiOverTwoRotation()) {
+        // -P'
+        result.value.sign = !result.value.sign;
+    }
+    else {
+        assert(this_op.hasPiOverFourRotation() &&
+               "Rotation of Clifford gate is neither pi/2 nor pi/4");
+        // P * P' * i
+        result = (*this->pauliString) * result * stim::FlexPauliString::from_text("i");
+    }
```

Comment:
I would recommend making the two Clifford cases explicit. This would keep the flow simple and encode the invariant.

```suggestion
    else if (this_op.hasPiOverFourRotation()) {
        // P * P' * i
        result = (*this->pauliString) * result * stim::FlexPauliString::from_text("i");
    }
    else {
       llvm_unreachable("Clifford rotation should be π/2 or π/4");
    }
```